src/os_crypto/shared/msgs.c

```diff
@@ -90,6 +92,7 @@ void OS_StartCounter(keystore *keys)
         }
 
         keys->keyentries[i]->fp = fopen(rids_file, "r+");
+        keys->keyentries[i]->inode = File_Inode(rids_file);
 
         /* If nothing is there, try to open as write only */
         if (!keys->keyentries[i]->fp) {
```

Comment:
Missing `File_Inode()` here, if `fopen()` failed. This call may be better after ending the `if` block (just like you did in `ReloadCounter()`.